csharp/src/Ice/Connection.cs

```diff
@@ -1052,12 +1049,20 @@ public override void Message(ref ThreadPoolCurrent current)
 
                                 byte major = _readStream.ReadByte();
                                 byte minor = _readStream.ReadByte();
-                                var pv = new Protocol(major, minor);
-                                Protocol.CheckSupportedProtocol(pv);
+                                if (major != 1 || minor != 0)
```

Comment:
Given that we do this check in at least 2 places, this could be a `CheckMagic` method defined on `Ice1Definitions`?

Reply:
Yes, will fix.

Reply:
Fixed with Ice1Definitions.CheckHeader that checks both Magic and ProtocolBytes and accepts a Span<byte>.  At a later point, we should read these header bytes directly as a span - there is no need to allocate an array for them.

Reply:
I was doing that as part of the InputStream updates
```
Span<byte> magic = _readBuffer.AsSpan(0, 4);
```
We my changes the creation of stream is delayed until we have read the size.